pymc3/distributions/transforms.py

```diff
@@ -499,6 +500,57 @@ def t_stick_breaking(eps: float) -> StickBreaking:
     return StickBreaking(eps)
 
 
+class StickBreaking2(Transform):
+    """
+    Transforms K - 1 dimensional simplex space (k values in [0,1] and that sum to 1) to a K - 1 vector of real values.
+    """
+
+    name = "stickbreaking"
+
+    def forward(self, x_):
+        x = x_.T
+        n = x.shape[0]
+        lx = tt.log(x)
+        shift = tt.sum(lx, 0, keepdims=True) / n
+        y = lx[:-1] - shift
+        return floatX(y.T)
+
+    def forward_val(self, x_):
+        x = x_.T
+        n = x.shape[0]
+        lx = np.log(x)
+        shift = np.sum(lx, 0, keepdims=True) / n
+        y = lx[:-1] - shift
+        return floatX(y.T)
+
+    def backward(self, y_):
+        y = y_.T
+        y = tt.concatenate([y, -tt.sum(y, 0, keepdims=True)])
+        # "softmax" with vector support and no deprication warning:
+        e_y = tt.exp(y - tt.max(y, 0, keepdims=True))
+        x = e_y / tt.sum(e_y, 0, keepdims=True)
+        return floatX(x.T)
+
+    def backward_val(self, y_):
+        y = y_.T
+        y = np.concatenate([y, -np.sum(y, 0, keepdims=True)])
+        x = np.exp(y)/np.sum(np.exp(y), 0, keepdims=True)
+        return floatX(x.T)
+
+    def jacobian_det(self, y_):
+        y = y_.T
+        Km1 = y.shape[0] + 1
+        sy = tt.sum(y, 0, keepdims=True)
+        r = tt.concatenate([y+sy, tt.zeros(sy.shape)])
+        # stable according to: http://deeplearning.net/software/theano_versions/0.9.X/NEWS.html
+        sr = tt.log(tt.sum(tt.exp(r), 0, keepdims=True))
+        d = tt.log(Km1) + (Km1*sy) - (Km1*sr)
```

Comment:
This is stable **after optimization**.  We're better off directly using the stable `Op` that replaces the unstable expression.

Reply:
I did not find a dedicated log-sum-exp operation in theano but I could use this: https://github.com/pymc-devs/pymc3/blob/67155fa4d000e73f99d761970ece73c2236a97d5/pymc3/math.py#L171

Reply:
[This](https://github.com/pymc-devs/Theano-PyMC/blob/master/theano/tensor/opt.py#L6772) is the optimization that Theano performs, which, yes, is pretty similar to `logsumexp`.

Reply:
@brandonwillard Thank you for the hint! I just used the existing pymc3 implementation so it can be changed centrally.